### ISSUE TITLE
How to enable polling after a producer is created with EventsPollingOption::Manual

### ISSUE THREAD

**Issue by YaoDing** [morganstanley/modern-cpp-kafka#126]
Hi,
It's kinda related to setErrorCallback/statsCallback, so if we create a producer with EventsPollingOption::Manual(because I want to set callbacks afterwards), it seems there isn't a public interface I can call to enable polling?

**Comment by kenneth-jia** [comment]
Hi, @YaoDing Here's a quite simple example for `pollEvents()`,  https://github.com/morganstanley/modern-cpp-kafka/blob/main/doc/KafkaProducerQuickStart.md#kafkaproducer-with-kafkaclientskafkaclienteventspollingoption
Generally speaking, you have to periodically call `producer.pollEvents()`, no matter whether there's any message just been sent or not.

**Comment by YaoDing** [comment]
Yeah, I was looking for something like startBackgroundPollingIfNecessary in Producer class so that I don't need to create thread/take care of it by myself :-). Given we allow users to create Producer with EventsPollingOption::Manual option, maybe you could consider make startBackgroundPollingIfNecessary public? But it's another topic, I will close this one now. Thank you for answering all my questions!

**Comment by kenneth-jia** [comment]
@YaoDing It doesn't make much sense to expose the `startBackgroundPollingIfNecessary` for users, -- it does nothing else except triggering the internal polling thread to start, then why not using the default producer constructor (with no `EventsPollingOption::Manual` specified)? While for a sophisticated user, the simple `pollEvents()` interface for a `manual polling` producer would be enough, -- we're able to create some "helper" code, but probably not what they want.
